number_test.go

```diff
@@ -584,3 +590,177 @@ func TestNumber_ConvertLesser(t *testing.T) {
 	value.chain.assertFailed(t)
 	value.chain.clearFailed()
 }
+
+func TestNumber_IsInt(t *testing.T) {
```

Comment:
Let's check how `IsInt()`, `NotInt()`, `IsInt(32)`, `NotInt(32)` work for the following boundary cases:

```go
-2^31 - 1
-2^31
-2^31 + 1
2^31 - 1
2^31
2^31 + 1
```

This would give us confidence that bitness checks are fully correct.

I suggest to replace subtests `without bit size` and `with bit size` with a single subtest that checks the values above.

We can also add a similar set of checks for other bitness, e.g. 16.

Reply:
Similar for IsUnit test.